### ISSUE TITLE
Question/Suggestion: New Potion Storage

### ISSUE THREAD

**Issue by Drukzah** [geheur/bank-tag-custom-layouts#99]
First time here so I wanted to keep it simple:

How likely and possible is it that the new potion storage system can be worked around (since it seems a bit clunky) so that bank tab layouts can still function the same way they do without it and let you withdraw the right pots (at least with the doses you have chosen in potion storage) from the storage directly, even with no placeholders in your main bank? Thanks

**Comment by geheur** [comment]
I don't have a potion storage and will not be getting one so if this is technically possible and allowed by the rules, someone else is going to have to do the work for this.

**Comment by Drukzah** [comment]
Fair. Ty for answering.

**Comment by dan-drews** [comment]
I'd be interested in trying to take this on. I saw Adam's change for adding this to the main client tab layouts. Is there any interaction between this plugin and the plugin built into runelite, or would it be a matter of re-implementing his changes?

**Comment by geheur** [comment]
There is no interaction between adamlayouts and bank tag layouts. They are mutually exclusive. So you would have to re-implement it.

**Comment by gtjamesa** [comment]
> I'd be interested in trying to take this on. I saw Adam's change for adding this to the main client tab layouts. Is there any interaction between this plugin and the plugin built into runelite, or would it be a matter of re-implementing his changes?

Just wondering how the progress on this going? 😊

**Comment by dan-drews** [comment]
I'm making progress on it, I'm fairly new to runelite plugins so still wrapping my head around some of the concepts. 

Rendering the potions with their counts is working (Screenshot below shows this, no potions in my true bank, only pot storage)
![image](https://github.com/user-attachments/assets/03055001-03c0-4e37-b048-f88d6c2bf8c0)

And pulling potions out of the bank is also working, for some reason it's only working if I don't select the first option in the list (the default left-click option) but it is working if I select any other item in the list
![image](https://github.com/user-attachments/assets/83de53a2-d86d-4295-b38b-728897eb3b6a)

Drag and drop of potions is not currently working either. All to say I'm making progress when I'm able to, but still have some oddities to get through

**Comment by gtjamesa** [comment]
Excellent work so far! I'm pretty new to plugin dev too but let me know if you want an additional tester

**Comment by dan-drews** [comment]
@gtjamesa That sounds great. I'll try to push my branch at some point when I think it's at least somewhat stable. Would love to get extra eyes to test it. I don't know all the use-cases or edge cases we could hit, or what other runelite/client settings could conflict with it so that'd be great. I use this plugin almost entirely via the Inventory Setup plugin, so a different perspective would be wonderful

**Comment by geheur** [comment]
@dan-drews btw if you are making this because of inventory setups you could also look at making inventory setups use core layouts to display items, instead.

**Comment by dan-drews** [comment]
Just an update here. The base functionality is working, and I fixed the above bug where left-click events were not being handled correctly.

I still have to fix drag and drop, which is pretty key functionality for this, and there's an issue where it's not properly caching the potion storage if you close and re-open the bank.

@geheur - I think I'd like to wrap this up and get it across the finish line, but I'd like to explore the option of adding it into the core layouts as well. That felt more daunting, and I think a good number of people use this combination so I wanted to help them out as well

**Comment by dan-drews** [comment]
@gtjamesa - I have the PR up, you're welcome to pull my branch and give it a try, let me know if any issues come up.

Note that you have to open your potion storage before these will populate in order to build the cache

**Comment by gtjamesa** [comment]
Hey,

The integration looks really great so far, and I can confirm that depositing/withdrawing potions is working as intended. I've noticed a couple of things that could hopefully be changed/added:

- The "default" item options such as duplicate layout are missing from the potion store potions. This doesn't present as a huge issue when using Inventory Setups because you'll rarely need to duplicate a potion, and pre-pots will be added as additional filtered items. However, some players may use the layouts plugin directly, and it would be impossible to add/remove additional potions. A good example of this is the popular "tob learner" layout
    ```
    banktaglayoutsplugin:tob - learner,14006:0,13072:1,15741:2,11665:3,11773:4,22948:5,8013:6,995:7,24780:8,13073:9,13239:10,21295:11,22322:12,23685:14,22400:15,22480:16,20997:17,12002:18,19547:19,22485:20,12926:21,555:22,13711:23,11663:24,11664:25,21791:26,28951:27,21003:28,27282:29,565:30,560:31,13441:32,23742:33,23694:34,13441:35,3046:36,12631:37,24777:38,29801:39,13441:40,3024:41,10588:42,3024:43,6685:44,13441:46,27282:47,13652:48,12695:49,23987:50,12695:51,6685:52,6685:54,12695:55,3024:56,6685:57,3024:58,3024:62,2444:63,12695:64,6685:65,2444:66,23987:80,11804:81,14728:82,1351:83,banktag:tob - learner,22473,3046,3024,995,27281,11773,23742,22400,13576,-11664,1351,19720,11804,13652,10588,21295,21791,21795,12899,12002,12006,8842,19553,19547,12954,12926,8013,11664,11663,11665,-13072,-13073,23694,23685,21003,22325,22323,22322,13441,29801,13072,13073,560,555,12631,12695,22947,566,565,24780,24777,13239,23987,2444,20997,6685,28951
     ```

  ![image](https://github.com/user-attachments/assets/c0d78ffa-cd60-444d-be76-7f6182872482)
  ![image](https://github.com/user-attachments/assets/ccf2445a-dea5-4dda-95de-569d73735a7a)
- As you pointed out, the potion quantities aren't displayed if the potion store hasn't yet been opened. This is because the potion caching is happening too early - I've fixed this on my local copy and submitted a PR ([#1](https://github.com/dan-drews/bank-tag-custom-layouts/pull/1))
- There's also the issue of adding potions to a tag/layout - do you know how the core adamlayouts is handling this? Prior to potion storage, you would have just dragged the potion into the side-tab
- I did find a random bug when adding a new additional filtered item (setups), removed it and it brought in some random potions. Still trying to reproduce 

*p.s. I was away this weekend - sorry for the delay!*

**Comment by dan-drews** [comment]
1. I'll take a look at that, I think I know where the issue is but I'm not sure. I also noticed that menu-entry swapper overrides are not respected, I'll want to take a closer look at that.
2. Awesome thank you! I'll get that reviewed when I have a chance, but this was a good find!
3. I think for AdamLayouts you have to have it in the bank, add it to the tag then put it in potion storage. Not the best workflow, but I'm not sure i'd be experienced enough to build out drag/drop from pot storage to tabs :(
4. Let me know if you can reproduce it. I really didn't look too much at modifying via inventory setups or via the plugin itself. I was mainly focused on the core functionality such as withdrawing and drag/drop. I should have probably focused on those other things too
5. I wonder if there's a way to override the default dosage in potion storage, and withdraw the actual number of doses that are tagged. I feel like if it were possible, Adam would have done that in his, maybe it's possible, but would be considered multiple actions?

**Comment by gtjamesa** [comment]
> I should have probably focused on those other things too

Not at all, that's what extra eyes are for! :)

> Adam would have done that in his, maybe it's possible, but would be considered multiple actions?

I did consider this too. I think that would be the ultimate solution and in my mind doesn't feel too far away from capturing a mouse click over a `FakeItem` and making it withdraw the real item. The difference I guess is that the dosage is supposed to be set per-potion. I'll have to carry on digging through the code to get more of a solid understand, but maybe @geheur can weigh in here?

**Comment by geheur** [comment]
My advice is to join runelite discord. I am not up to date with either a) potion storage stuff and b) inventory setups upcoming changes to do the layouting without BTL.

**Comment by dan-drews** [comment]
Based on some conversation in the runelite discord it sounds like the maintainer of inventory setups is integrating that with adamlayouts.

https://github.com/dillydill123/inventory-setups/issues/277 They're also discussing it here. I think it's probably worth holding off on this until we know how that turns out, personally.

**Comment by dan-drews** [comment]
Okay, the maintainer over there said it'd be at least a couple of weeks to get that integrated

**Comment by gtjamesa** [comment]
> I think it's probably worth holding off on this until we know how that turns out, personally.

I do see what you mean from the perspective of an integration with Inventory Setups, but AFAIK, this plugin isn't being retired, so it is probably worth getting the PR to a mergeable state?

**Comment by dan-drews** [comment]
@gtjamesa  yeah, you're right. I haven't had much time to work on it this past weekend, but we'll see if i can get to it during this week. Unless @geheur advises against this. Do you plan to maintain this longer-term after adamlayouts starts to support inventory setups? Are there any other feature differences between the 2 plugins?

**Comment by geheur** [comment]
I don't have any plants to continue developing any of my plugins atm.

As for future utility of this plugin... that's for other people to determine. I have not used adamlayouts much (I no longer play the game) so I can't comment on any notable differences besides that adamlayouts doesn't limit the scroll height of the tag tab. If someone sees value and wants to continue development on hub BTL that is fine but I don't plan to put much time into it personally.

**Comment by dillydill123** [comment]
Just commenting here that I've finished the integration of core bank tag layouts with inventory setups (which should support potion storage) and could use some help testing. If you'd like to help, please take a look here: https://github.com/dillydill123/inventory-setups/issues/277#issuecomment-2439334648

**Comment by kieranjen** [comment]
@dan-drews hey! I don't suppose you've made any further progress on this?

**Comment by dan-drews** [comment]
@kieranjen I did not complete this, since the built-in bank tag layouts was able to solve my problem. I'm not sure if there's an easy way to migrate from this plugin to the one built into runelite but if there is a way, that may be teh best approach
